src/Microsoft.Extensions.Configuration.Abstractions/IConfigurationSection.cs

```diff
@@ -1,13 +1,27 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-
 namespace Microsoft.Extensions.Configuration
 {
+    /// <summary>
+    /// REVIEW: Really not confident these are correct.
+    /// Represents a sub-section of application configuration values.
+    /// </summary>
     public interface IConfigurationSection : IConfiguration
```

Comment:
To help document these here is their usage in the two primary scenarios that a user would access Key/Path/Value properties. 

```cs
var config = new ConfigurationBuilder().AddXxxFile("sample.xxx").Build();

// Loop through child objects in a section
// Key is the item's 'name'
// Path is the full string to the item and always ends with Key
foreach(var item in config.GetSection("fruits").GetChildren()))
{
   var key = item.Key;   // "Apple", "Avocado"
   var color = item["color"]; // "Red", "Yellow"
   var shape = item["shape"]; // "Round", "Oblong"

   var path = item.Path;   // "fruits:Apple", "fruits:Avocado"
   var val = item.Value;   // empty string - these paths do not have values
}

// Loop through child values in an array
// Key is the item's 'index'
// Path is the full string to the item and always ends with Key
// Value is the simplex value
foreach(var item in config.GetSection("sizes").GetChildren()))
{
   var key = item.Key;   // "0", "1", "2"
   var val = item.Value;   // "S", "M", "L"

   var path = item.Path;   // "sizes:0", "sizes:1", "sizes:2"
}
```
## Sample files for this code
```json
{
  "fruits" : {
    "Apple": {
      "color": "Red",
      "shape": "Round"
    },
    "Avocado": {
      "color": "Yellow",
      "shape": "Oblong"
    }
  },
  "sizes": ["S", "M", "L"]
}
```
```ini
[fruits:Apple]
color=Red
color=Round
[fruits:Avocado]
color=Yellow
color=Oblong
#or
[fruits]
Apple:color=Red
Apple:shape=Round
Avocado:color=Yellow
Avocado:shape=Oblong

[sizes]
0=S
1=M
2=L
```
```xml
<config>
  <fruits name="Apple">
    <color>Red</color>
    <shape>Round</shape>
  </fruits>
  <fruits name="Avocado">
    <color>Yellow</color>
    <shape>Oblong</shape>
  </fruits>
  <!-- or -->
  <fruits>
    <Apple color="Red" shape="Round"/>
    <Avocado color="Yellow" shape="Oblong"/>
  </fruits>
  <!-- or -->
  <fruits name="Apple" color="Red" shape="Round" />
  <fruits name="Avocado" color="Yellow" shape="Oblong" />

  <sizes name="0">S</sizes>
  <sizes name="1">M</sizes>
  <sizes name="2">L</sizes>
</config>
```
```batch
SET FRUITS__APPLE__COLOR=Red
SET FRUITS__APPLE__SHAPE=Round
SET FRUITS__AVOCADO__COLOR=Yellow
SET FRUITS__AVOCADO__SHAPE=Oblong
SET SIZES__0=S
SET SIZES__1=M
SET SIZES__2=L
```